extras/bindings/c/src/SolverInterfaceC.cpp

```diff
@@ -1,41 +1,44 @@
 extern "C" {
 #include "precice/SolverInterfaceC.h"
 }
+#include <memory>
 #include <string>
 #include "precice/SolverInterface.hpp"
 #include "precice/impl/versions.hpp"
 #include "utils/assertion.hpp"
 
-static precice::SolverInterface *interface = nullptr;
+static std::unique_ptr<precice::SolverInterface> interface = nullptr;
 
-void precicec_createSolverInterface(
+void precicec_createSolverInterface_withCommunicator(
     const char *participantName,
     const char *configFileName,
     int         solverProcessIndex,
-    int         solverProcessSize)
+    int         solverProcessSize,
+    void *      communicator)
 {
   std::string stringAccessorName(participantName);
   std::string stringConfigFileName(configFileName);
-  interface = new precice::SolverInterface(stringAccessorName,
-                                           stringConfigFileName,
-                                           solverProcessIndex,
-                                           solverProcessSize);
+
+  PRECICE_ASSERT(interface == nullptr);
```

Comment:
As this is covers wrong API usage, this should be a `PRECICE_CHECK`.
We use assertions only for checking internal pre/post conditions. A user _should_ never run into a `PRECICE_ASSERT`.

Using `CHECK` requires a logger. You can have a look at the fortran bindings.
https://github.com/precice/precice/blob/583339a950427b530bc4872649145e14f66cb0ea/extras/bindings/fortran/src/SolverInterfaceFortran.cpp#L14

Reply:
How should this be changed? The solver interface for C only uses `PRECICE_ASSERT` and not a single `PRECICE_CHECK`. Should I replace all assertions by checks?

Reply:
In my opinion this could also be solved in another PR. Not related to the main theme here.

Reply:
I have just changed `PRECICE_ASSERT` to `PRECICE_CHECK` and the name of the `unique_ptr` to `impl` in 04a637e156673b42032993697f53b7ad87f4a1b1. If you do not like that, I would suggest to drop the commit.